tests/subsys/rtio/rtio_i2c/include/blocking_emul.hpp

```diff
@@ -0,0 +1,15 @@
+/* Copyright (c) 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <functional>
+
+#define CUSTOM_FFF_FUNCTION_TEMPLATE(RETURN, FUNCNAME, ...)                                        \
+	std::function<RETURN(__VA_ARGS__)> FUNCNAME
```

Comment:
I added a comment for why this makes the tests much more readable the TL;DR is that we can keep all the values local to the test function instead of making them global and having to jump around when trying to debug the test.